### PR TITLE
archive: allow ZIP64 by default

### LINKED ISSUE #24125: Archive is unable to create zip archives over 4GB
##### ISSUE TYPE

 - Bug Report
 
##### COMPONENT NAME

archive module

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = /home/user/projectpath/ansible.cfg
  configured module search path - Default w/o overrides
  python version - 2.7.5 (default, Oct 11 2015, 17:47:16) [GCC 4.8.3 20140911 (Red Hat 4.8.3-9)]

```

##### OS / ENVIRONMENT

N/A

##### SUMMARY

When trying to create a zip archive with a files over 4GB in size the task fails stating
the following:

Filesize would require ZIP64 extensions

##### STEPS TO REPRODUCE

Create a file over 4GB

$ truncate 5GB test.file

```yaml
- name: Archive test
  hosts: localhost

  tasks:
    - archive:
         path: test.file
         dest: test.file.zip
         format: zip
```

##### EXPECTED RESULTS

A zip archive named test.file.zip is created containing the file test.file

##### ACTUAL RESULTS

```
fatal: [hostname]: Failed! => {"changed": false, "failed": true, "msg": "Error when writing zip at test.file.zip: Filesize would require ZIP64 extensions"}
```

### PR DESCRIPTION
**Pull request by fridim**
##### SUMMARY
Fixes #24125

allowZip64 is True by default since python 3.4,
see https://docs.python.org/3/library/zipfile.html#zipfile-objects

With python 2.7, when the file or folder to zip is >2G,
zipfile raises LargeZipFile.

This commit set allowZip64 to true.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
modules/files/archive.py

##### ANSIBLE VERSION

```
ansible 2.4.0 (issue-24125 ddbe9efe36) last updated 2017/05/02 13:21:03 (GMT +200)
  config file =
  configured module search path = [u'/home/fridim/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/fridim/sync/dev/ansible/lib/ansible
  executable location = /home/fridim/sync/dev/ansible/bin/ansible
  python version = 2.7.13 (default, Feb 11 2017, 12:22:40) [GCC 6.3.1 20170109]

```

##### ADDITIONAL INFORMATION
Currently, with python < 3.4, zipfile will raise if file is large because zip needs ZIP64 extension.

With this patch, at least it will try to add zip64 extension when needed.

### CODE REVIEW

**Comment by robinro** [comment]
zip64 is not supported for old unzip versions like in rhel 5: http://h20564.www2.hpe.com/hpsc/doc/public/display?docId=emr_na-c02251051

**Comment by fridim** [comment]
@robinro thanks for the link. My idea was to allow always, since it is used only when needed. But you have a point, it has to be tested on those systems not supporting zip64, in order to return the proper error.

(I don't have a rhel5 env ready yet, but i'll try to set one up. Do you know what happens with the allowzip64 set to true on those systems not supporting zip64? Something we could easily catch maybe ?)

Currently allowzip64 param value is bound to python version, but it should rather be set to true/false depending on the unzip version instead. 

With current code,  you can fall in the following situation :
- python 2
- unzip supporting zip64 but unable to use it with ansible

**Comment by sivel** [comment]
Ansible 2.4 is dropping support for python 2.4, so I don't think that is an issue.

**Comment by fridim** [comment]
Indeed tried to run a simple playbook with only archive module, on rhel5, it's not even running:

```
fatal: [192.168.122.28]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "module_stderr": "Shared connection to 192.168.122.28 closed.\r\n", 
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_85lV2y/ansible_module_archive.py\", line 126, in ?\r\n    from ansible.module_utils.basic im
port AnsibleModule\r\n  File \"/tmp/ansible_85lV2y/ansible_modlib.zip/ansible/module_utils/basic.py\", line 164, in ?\r\n  File \"/tmp/ansible_85lV2y/ansible_modlib.zip/ansib
le/module_utils/pycompat24.py\", line 55, in ?\r\n  File \"/tmp/ansible_85lV2y/ansible_modlib.zip/ansible/module_utils/six/__init__.py\", line 61\r\n    from . import _six as
 _bundled_six\r\n         ^\r\nSyntaxError: invalid syntax\r\n", 
    "msg": "MODULE FAILURE", 
    "rc": 0
}
```

**Comment by fridim** [comment]
> Ansible 2.4 is dropping support for python 2.4, so I don't think that is an issue.

@sivel @robinro so, are we safe to set allowZip64 to true ?

**Review by robinro — APPROVED** [review]
I'm fine with `allowZip64=True`.
LGTM

**Comment by fridim** [comment]
@robinro are we waiting for someone else to review?

**Comment by robinro** [comment]
bot_status

**Comment by robinro** [comment]
@bendoh Please review.

@fridim If @bendoh, the maintainer of this module, does not respond soon, also someone from code could merge it.

@sivel You already had a look. Can you review/merge this?

**Comment by calfonso** [comment]
rebuild_merge

**Pull request merged**
